### PR TITLE
Fixes #948

### LINKED ISSUE #948: SG as a service is not supported on Windows
believe that we need to have the same functionality on all supported platforms

for linux:
https://github.com/couchbase/sync_gateway/tree/master/service

### PR DESCRIPTION
**Pull request by zgramana**
Implements a simple Windows service wrapper around `sync_gateway.exe`.
This is now be installed as a service using the command:

```
sc create SyncGateway binPath=
"C:\sources\sync_gateway\service\sg-windows\sg-windows.exe \"C:\Program
Files (x86)\Couchbase\examples\basic-couchbase-bucket.json\""
DisplayName= "Couchbase Sync Gateway"
```

> **NOTE**: the path to `sg-windows.exe` will need to be changed in an actual
> deployment.

Whatever config JSON file path is passed to `sg-windows.exe` will be in turn
passed to `sync_gateway.exe`.

In this commit, sync_gateway.exe is expect to be located at `C:\Program
Files (x86)\Couchbase\sync_gateway.exe`.

### CODE REVIEW

File: service/sg-windows/sgservice.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	completed := make(chan error, 1)
+	go func() {
+		completed <- p.SyncGateway.Wait()
+	}()
+	select {
+	case err := <- completed:
+		if err != nil {
+			logger.Errorf("Sync Gateway exitting with status = %v", err)
+		}
+	}
```
Comment (tleyden) [inline service/sg-windows/sgservice.go:40]: I think the goroutine + channel + select here isn't doing anything useful.  

Eg, replace:

```
completed := make(chan error, 1)
+   go func() {
+       completed <- p.SyncGateway.Wait()
+   }()
+   select {
+   case err := <- completed:
+       if err != nil {
+           logger.Errorf("Sync Gateway exitting with status = %v", err)
+       }
+   }
```

with: 

```
err := p.SyncGateway.Wait()
```

File: service/sg-windows/sgservice.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+	"github.com/kardianos/service"
+	"os"
+)
+
+var logger service.Logger
+
+type program struct{
+	ExePath 	string
+	ConfigPath  string
```
Comment (tleyden) [inline service/sg-windows/sgservice.go:16]: The types should line up vertically, looks like this wasn't run through `gofmt`yet?

File: service/sg-windows/sgservice.go (hunk 1 above)
Comment (zgramana) [inline service/sg-windows/sgservice.go:40]: Cheers. Pushed the change, along with convenience support for installing it as a Windows service, as well as basic operations like start, stop, restart, and uninstall.

File: service/sg-windows/sgservice.go (hunk 2 above)
Comment (zgramana) [inline service/sg-windows/sgservice.go:16]: done and pushed.

**Pull request merged**